internal/server/statvar/fetcher/formula_fetcher.go

```diff
@@ -39,83 +41,108 @@ func FetchFormulas(
 	remoteRespChan := make(chan *pbv2.NodeResponse, 1)
 	// Fetch for BT and SQL.
 	errGroup.Go(func() error {
-		data, _, err := v1pv.Fetch(
-			errCtx,
-			store,
-			[]string{"StatisticalCalculation"},
-			[]string{"typeOf"},
-			0,
-			"",
-			"in",
-		)
-		if err != nil {
-			return err
-		}
 		statCal := []string{}
-		for _, nodes := range data["StatisticalCalculation"]["typeOf"] {
-			for _, node := range nodes {
-				statCal = append(statCal, node.Dcid)
+		nextToken := ""
+		for ok := true; ok; ok = (nextToken != "") {
+			statCalReq := &pbv1.TriplesRequest{
+				Node:      "StatisticalCalculation",
+				Direction: "in",
+				NextToken: nextToken,
+			}
+			statCalResp, err := triples.Triples(
+				errCtx,
+				statCalReq,
+				store,
+				metadata,
+			)
+			if err != nil {
+				return err
+			}
+			typeOf, ok := statCalResp.Triples["typeOf"]
+			if ok {
+				for _, node := range typeOf.Nodes {
+					statCal = append(statCal, node.Dcid)
+				}
 			}
+			nextToken = statCalResp.GetNextToken()
 		}
 		if len(statCal) == 0 {
 			return nil
 		}
-		resp, _, err := v1pv.Fetch(
-			errCtx,
-			store,
-			statCal,
-			[]string{"outputProperty", "inputPropertyExpression"},
-			0,
-			"",
-			"out",
-		)
-		if err != nil {
-			return err
-		}
-		// Wrap result in pbv2.NodeResponse.
 		localResp := &pbv2.NodeResponse{Data: map[string]*pbv2.LinkedGraph{}}
-		for dcid, data := range resp {
-			localResp.Data[dcid] = &pbv2.LinkedGraph{
-				Arcs: map[string]*pbv2.Nodes{},
+		for ok := true; ok; ok = (nextToken != "") {
+			bulkReq := &pbv1.BulkTriplesRequest{
+				Nodes:     statCal,
```

Comment:
Would this scale if we have millions or even 10s of thousands of stat cals?

Actually, can you help me understand the overall context of where and how the formula fetcher is used? e.g. is it used by external clients, internally, is it called once and cached, etc.

Reply:
Spoke offline: this is loaded once and stored in Mixer cache